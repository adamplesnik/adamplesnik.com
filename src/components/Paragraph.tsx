import clsx from 'clsx'
import { HTMLAttributes } from 'react'

const Paragraph = ({ children, className }: HTMLAttributes<HTMLParagraphElement>) => {
  return <p className={clsx('w-full max-w-xl leading-relaxed', className)}>{children}</p>
}

export default Paragraph
