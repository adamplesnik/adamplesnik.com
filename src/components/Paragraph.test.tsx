import { toJson } from '@/utils/toJson'
import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'
import Paragraph from './Paragraph'

test('Small paragraph', () => {
  const component = renderer.create(<Paragraph size="small">Small paragraph</Paragraph>)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Regular paragraph', () => {
  const component = renderer.create(<Paragraph size="regular">Regular paragraph</Paragraph>)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Large paragraph', () => {
  const component = renderer.create(<Paragraph size="large">Large paragraph</Paragraph>)
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Regular paragraph with class name', () => {
  const component = renderer.create(
    <Paragraph size="regular" className="test-class">
      Regular paragraph with class name
    </Paragraph>
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
