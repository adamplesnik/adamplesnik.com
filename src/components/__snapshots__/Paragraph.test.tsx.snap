// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Regular paragraph 1`] = `
<p
  className="mb-4 w-full text-base leading-7 last:mb-0"
>
  Regular paragraph
</p>
`;

exports[`Regular paragraph with class name 1`] = `
<p
  className="mb-4 w-full text-base leading-7 last:mb-0 test-class"
>
  Paragraph with class name
</p>
`;
